[FE] - random unit test à fixer

## Story:

### Description

Résoudre le test unitaire suivant :

```
 FAIL  src/features/student/declaredSkills/components/overlays/modals/AssociateActivitiesToDeclaredSkillModal/AssociateActivitiesToDeclaredSkillModal.test.ts > 🔵 GIVEN an associate activities to declared skill modal > 🔶 WHEN loading activities fails > 🟩 THEN it should not add a success toaster message
AssertionError: expected "spy" to not be called at all, but actually been called 1 times

Received: 

  1st spy call:

    Array [
      Object {
        "description": "Vous avez associé 2 activités. Retrouvez-les dans la catégorie "Mes activités associées".",
        "timeout": 2000,
      },
    ]


Number of calls: 1

 ❯ src/features/student/declaredSkills/components/overlays/modals/AssociateActivitiesToDeclaredSkillModal/AssociateActivitiesToDeclaredSkillModal.test.ts:170:41
    168| 
    169|     BddTest().then('it should not add a success toaster message', () =…
    170|       expect(mockAddSuccessMessage).not.toHaveBeenCalled()
       |                                         ^
    171|     })
    172| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯


 Test Files  1 failed | 432 passed (433)
      Tests  1 failed | 4661 passed (4662)
```

### Éléments UI de travail

[lien Figma](URL_figma)

<!-- insérer une image -->

--- a/src/features/student/declaredSkills/components/overlays/modals/AssociateActivitiesToDeclaredSkillModal/AssociateActivitiesToDeclaredSkillModal.test.ts
+++ b/src/features/student/declaredSkills/components/overlays/modals/AssociateActivitiesToDeclaredSkillModal/AssociateActivitiesToDeclaredSkillModal.test.ts
@@ -1,3 +1,4 @@
+import type { AssociationActivity } from '@/features/student/traces/views/StudentTraceView/components/overlays/modals/AssociateActivitiesModal/AssociateActivitiesModal.vue'
 import type { VueWrapper } from '@vue/test-utils'
 import {
   associateDeclaredSkillWithDeclaredActivityErrorHandler,
@@ -75,16 +76,18 @@ BddTest().given('an associate activities to declared skill modal', () => {
     })
 
     BddTest().and('the inner modal emits search event', () => {
+      let initialActivities: AssociationActivity[]
+
       beforeEach(async () => {
+        initialActivities = modal.props('activities') as AssociationActivity[]
         modal.vm.$emit('search', 'valeurs')
         await flushPromises()
       })
 
-      BddTest().then('it should filter activities through the query', async () => {
+      BddTest().then('it should update activities after search', async () => {
         await vi.waitFor(() => {
-          const updatedModal = wrapper.findComponent(AssociateActivitiesModalStub) as VueWrapper<InstanceType<typeof AssociateActivitiesModalStub>>
-          const activities = updatedModal.props('activities')
-          expect(activities.length).toBeGreaterThanOrEqual(0)
+          const updatedModal = wrapper.findComponent(AssociateActivitiesModalStub)
+          expect(updatedModal.props('activities')).not.toEqual(initialActivities)
         })
       })
     })
@@ -97,21 +100,19 @@ BddTest().given('an associate activities to declared skill modal', () => {
         })
 
         modal.vm.$emit('associate', ['activity-search-1', 'activity-search-2'])
-        await flushPromises()
-      })
-
-      BddTest().then('it should emit associated event', async () => {
         await vi.waitFor(() => {
           expect(wrapper.emitted('associated')).toBeTruthy()
         })
       })
 
-      BddTest().then('it should show a success toaster with the correct count', async () => {
-        await vi.waitFor(() => {
-          expect(mockAddSuccessMessage).toHaveBeenCalledWith({
-            timeout: 2000,
-            description: expect.stringContaining('2')
-          })
+      BddTest().then('it should emit associated event', () => {
+        expect(wrapper.emitted('associated')).toHaveLength(1)
+      })
+
+      BddTest().then('it should show a success toaster with the correct count', () => {
+        expect(mockAddSuccessMessage).toHaveBeenCalledWith({
+          timeout: 2000,
+          description: expect.stringContaining('2')
         })
       })
 
@@ -126,7 +127,7 @@ BddTest().given('an associate activities to declared skill modal', () => {
       })
 
       BddTest().then('it should emit cancel', () => {
-        expect(wrapper.emitted('cancel')).toBeTruthy()
+        expect(wrapper.emitted('cancel')).toHaveLength(1)
       })
     })
   })

